### ISSUE TITLE
Upgrade is not purchaseable

### ISSUE THREAD

**Issue by Morn2** [Infiland/AsteroidsFeedback#39]
The "Machine Gun" is currently on Lvl 11 and the ring around it is green. There is the turning coins and a price of 941 visible. 
I have 1048$, so it should be possible to buy the upgrade. But nothing happens when i click on it. 

When i compare it to as example the "Mini Rocket Pots". There i see also the green ring around a 7, but no coins, so i know that i cant buy an upgrade at this stage. 

I had the same issue on an earlier level of the machine, it was buy-able then after i think some level ups. 

I play on Release 1.2.4a which i have from steam. I run it on Manjaro Linux and can show you more specific info if you wish.

So this could be just a cosmetic issue, or it is a bug. I thought it is good when i reporting it. I really like this game, keep up with this great work :)

![Asteroids_Machine_Gun_no_update_possible](https://github.com/user-attachments/assets/43086be9-c021-4d30-952a-59780bf7b1a0)

**Comment by Morn2** [comment]
Sorry, i was probably a bit too fast. I discovered now, that i have an EXP Bar for the Machine Gun. Is it possible that the Update can only be started, when the exp. are full? Then, it would be only a cosmetic problem and could easy be fixed with like an extra bar on the icon itself or something that shows the player, that he needs to gain for exp for this weapon. 

![Asteroids Machine Gun](https://github.com/user-attachments/assets/eea507ae-5826-4589-8196-f3ec38410590)

**Comment by Infiland** [comment]
Can I ask how much EXP do you have for the rifle (or Mini Rocket Pots). It could be a visual glitch actually because as soon as you've reached the cap then you should be able to have level 12 unlocked, but this will be fixed in 1.2.5

**Comment by Infiland** [comment]
Also thanks for liking my game :)

**Comment by Morn2** [comment]
I started the game again after i played a bit, so the EXP could be more. The machine gun is level 12 now, the coin is still there but the $ amount is away. 

I made some screenshots so you can check all out:

![Machine_Gun_EXP](https://github.com/user-attachments/assets/db32c46a-6921-4a42-90c8-394c3fc488ec)
![Mini_Rock_Pods](https://github.com/user-attachments/assets/e878f5df-3d53-4dd8-88d1-3d463b9f2e36)

Also a visual glitch could be that the Ability is shown on the main menu in the corner down left. This gives interferences in some sub menus. 

![Main_Menu](https://github.com/user-attachments/assets/b18ef45b-ed02-4678-9c90-340051e6d57d)


That is no problem, i really love this game and when i have some time, i could also help with a German translation. I will come to the discord server when i have time.

**Comment by Infiland** [comment]
Yes, the coin icon is not supposed to be there when an shop item is level 12, and I will hide the ability hud in main menu

**Comment by Infiland** [comment]
Fixed in 1.4.1
